### ISSUE TITLE
Word Wrap is imprecise with non-monospace fonts

### ISSUE THREAD

**Issue by jmcooper** [microsoft/vscode#12421]
I'm seeing a wide right margin in my  editor when I have word wrap on.
![widemargin](https://cloud.githubusercontent.com/assets/1505427/18737217/0f1066ca-804c-11e6-9bd8-281f0a56fe67.PNG)
I don't see this on other computers.  Not sure what's up.  This is all I have in my user settings:
// Place your settings in this file to overwrite the default settings
{
 "editor.tabSize": 2,
 "editor.insertSpaces": true,
 "editor.fontSize": 21,
 "editor.fontFamily": "Source Code Pro",
 "git.enabled": false
}
- VSCode Version: Code 1.5.2 (66f37fd2a99eb9d628dd374d81d78835b410c39b, 2016-09-12T13:30:03.456Z)
- OS Version: Windows_NT ia32 10.0.10586

Steps to Reproduce:
1. Turn on Word Wrap (alt-z)

**Comment by alexandrudima** [comment]
@jmcooper Try changing the setting: `editor.wrappingIndent: "none"`. It is `"same"` by default

**Comment by jmcooper** [comment]
@alexandrudima Thanks, but that affects the behavior of the left indent.  I've tried changing that and it doesn't help.

**Comment by jmcooper** [comment]
@chrmarti Is there some info you need that I can get you?

**Comment by chrmarti** [comment]
@jmcooper To clarify: You are referring to the spacing _left_ of the line numbers, right?

**Comment by jmcooper** [comment]
No, I'm referring to all the white space on right side, after the code. It
is like the code is wrapping way too soon.

**Comment by chrmarti** [comment]
We only wrap on whitespace it seems.

**Comment by jmcooper** [comment]
That's not the problem, I already tested that by adding some whitespace and
it still wrapped way early.

**Comment by jmcooper** [comment]
Notice in this screenshot how line 22 is wrapping.  There is plenty of space for the `});` and yet it wrapped because of some invisible, wide margin on the right. If I make my window wider it will unwrap, so it does appear to be a wrapping problem.

![widemargin2](https://cloud.githubusercontent.com/assets/1505427/18810702/366a5efa-825a-11e6-9449-fa8f4b40e90c.PNG)

**Comment by alexandrudima** [comment]
@jmcooper Thank you for the clarifications. Now I understand what you mean.

Word wrapping cut points are computed on our side based on two characters widths, namely the width of `n` for half-width characters and the width of `ｍ` for CJK full-width characters.

Given your text doesn't contain CJK full-width characters, only the width of `n` is used. So each character counts as the width of `n`, including `i` or `j` which are quite thin.

TL;DR The computation of wrapping cut points is imprecise for non-monospaced fonts.

**Comment by jmcooper** [comment]
Ahh, this makes sense. I didn't realize the font I was trying to use wasn't installed, so I was getting a non-monospaced font.

**Comment by alexdima** [comment]
Let's track in #2654
